src/main/java/org/jitsi/jicofo/FocusManager.java

```diff
@@ -360,14 +363,18 @@ private void createConference(
             = JitsiMeetGlobalConfig.getGlobalConfig(
                 FocusBundleActivator.bundleContext);
 
-        String id = generateConferenceId();
-        JitsiMeetConferenceImpl conference
-            = new JitsiMeetConferenceImpl(
+        JitsiMeetConferenceImpl conference;
+        String id;
+        synchronized (conferencesSyncRoot)
+        {
+            id = generateConferenceId();
```

Comment:
Yeah, I agree it could be done this way, and I can change it if you think that's a better way. I prefer the current code, because it obtains the lock once (when the second synchronized block is reached the thread already has the lock so it doesn't block) as opposed to obtaining it, then releasing it and then obtaining again. I don't know if that line of reasoning is correct.